dockstore-webservice/src/test/java/io/dockstore/webservice/helpers/DiagnosticsHelperTest.java

```diff
@@ -0,0 +1,140 @@
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import java.util.Random;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+import org.slf4j.event.Level;
+import org.slf4j.helpers.AbstractLogger;
+
+class DiagnosticsHelperTest {
+
+    private static final String BASE64_CHARS = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ/+";
+    private static DiagnosticsHelper helper;
+    private static String output;
+
+    @BeforeAll
+    static void setup() {
+        Logger logger = new AbstractLogger() {
+            @Override
+            protected void handleNormalizedLoggingCall(Level level, Marker marker, String messagePattern, Object[] arguments, Throwable throwable) {
+                if (arguments != null && arguments.length > 0) {
+                    throw new RuntimeException("unexpected arguments");
+                }
+                if (throwable != null) {
+                    throw new RuntimeException("unexpected throwable");
+                }
+                output = messagePattern;
+            }
+            @Override
+            protected String getFullyQualifiedCallerName() {
+                return "";
+            }
+            @Override
+            public boolean isErrorEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isErrorEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isWarnEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isWarnEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isInfoEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isInfoEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isDebugEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isDebugEnabled() {
+                return true;
+            }
+            @Override
+            public boolean isTraceEnabled(Marker marker) {
+                return true;
+            }
+            @Override
+            public boolean isTraceEnabled() {
+                return true;
+            }
+        };
+        helper = new DiagnosticsHelper(logger);
+    }
+
+    @Test
+    void testCensoringWithExamples() {
+        confirmPassed("/This/Is/A/Test/Of/The/Censoring/System");
+        confirmPassed("a short decimal number: 7236463, a short hex number af081afb609d");
+        confirmCensored("f893b559e572f323b5369452752d56743970e39132136aff378b76492c7549dd");
+        confirmCensored("039832784664398210947231904282437420197091837413209843");
```

Comment:
All of the key/credential/secret-like strings in this file are fake, I created them by banging on the keyboard and or encoding `/dev/random`.